routes/rooms.go

```diff
@@ -56,6 +57,50 @@ func GetEmergencyRoomById(ctx *gin.Context) {
 	ctx.JSON(http.StatusOK, resp)
 }
 
+// GetEmergencyRooms godoc
+// @Summary    get all emergency rooms
+// @Tags       emergency-rooms
+// @Produce    json
+// @Param      page                            query       uint                    false   "0-indexed page number, 0 is assumed when omitted"
+// @Success    200                             {object}    GetMultipleERsResponse
+// @Failure    501                             {object}    HTTPErrorResponse
+// @Router     /er                             [get]
+
+func Paginate(r *http.Request) func(db *gorm.DB) *gorm.DB {
+	return func(db *gorm.DB) *gorm.DB {
+		q := r.URL.Query()
+		page, _ := strconv.Atoi(q.Get("page"))
+		if page == 0 {
+			page = 1
+		}
+
+		pageSize, _ := strconv.Atoi(q.Get("page_size"))
+		switch {
+		case pageSize > 100:
+			pageSize = 100
+		case pageSize <= 0:
+			pageSize = 10
+		}
+
+		offset := (page - 1) * pageSize
+		return db.Offset(offset).Limit(pageSize)
+	}
+}
+
+func GetEmergencyRooms(ctx *gin.Context) {
+	_, logCtx := logging.GetRequestLogger(ctx)
+	db := models.GetDB(logCtx)
+
+	var emergencyRooms []models.EmergencyRoom
+	dbResult, _ := db.Scopes(Paginate(ctx.Request)).Find(&emergencyRooms).Rows()
+
+	for dbResult.NextResultSet() {
+		emergencyRooms = append(emergencyRooms, models.EmergencyRoom{})
```

Comment:
As long as there is not go magic going on that I don't understand, this adds empty ER structs to the array, right?

Note:
The EmergencyRoom struct also contains many to many references about Departments, that we don't want to return because they themselves reference the EmergencyRoom struct (recursion). Use `models.DepartmentsToBases(er.Departments)` to convert it to a basic version without those references and build `GetSingleERResponse`s like here: https://github.com/helpwave/rest-api/blob/main/routes/rooms.go#L107 (just in a loop)